cmd/validate_image.go

```diff
@@ -23,15 +23,17 @@ import (
 	"io/ioutil"
 	"sync"
 
+	ecc "github.com/hacbs-contract/enterprise-contract-controller/api/v1alpha1"
 	"github.com/hashicorp/go-multierror"
 	appstudioshared "github.com/redhat-appstudio/managed-gitops/appstudio-shared/apis/appstudio.redhat.com/v1alpha1"
 	"github.com/spf13/cobra"
 
 	"github.com/hacbs-contract/ec-cli/internal/applicationsnapshot"
 	"github.com/hacbs-contract/ec-cli/internal/output"
+	"github.com/hacbs-contract/ec-cli/internal/policy"
 )
 
-type imageValidationFunc func(ctx context.Context, imageRef, policyConfiguration, publicKey, rekorURL string) (*output.Output, error)
+type imageValidationFunc func(ctx context.Context, imageRef string, policy *ecc.EnterpriseContractPolicySpec) (*output.Output, error)
```

Comment:
For further refactor (another PR), we could instead of `ecc.EnterpriseContractPolicySpec` have a ec-cli or ec-policies configuration object here and map the `ecc.EnterpriseContractPolicySpec` to it instead, to have a intermediate common layer so to say...

Reply:
:+1:  I like this idea. Let's do that on a follow up PR.